### PR TITLE
changelog: Add details for cache speedup and performance improvements

### PR DESCRIPTION
**Pull request by tmessi**
Refs: f431d1e22309609baf37333b89d7b341141757a3
Refs: d5817f0adab7f5c1f045fa4a5c15a6dc553b022c

### CODE REVIEW

**Review by mgaffney — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,12 +4,44 @@ Canonical reference for changes, improvements, and bugfixes for Boundary.
...
+  deletes/updates to session tables.
+  ([PR](https://github.com/hashicorp/boundary/pull/5128))
+* Reorder indexes on several join tables to improve performance of grants query.
+  ([PR](https://github.com/hashicorp/boundary/pull/5128))
+* Make client cache sqlite database persistent between restarts of the client
+  cache daemon. ([PR](https://github.com/hashicorp/boundary/pull/5128))
+* Improve client cache performance by adding indexes, limiting results,
+  and insuring only one refresh is running at a time for a given user and
+  resource. ([PR](https://github.com/hashicorp/boundary/pull/5128))
+
```
Comment (jefferai) [inline CHANGELOG.md:21]: It's missing https://github.com/hashicorp/boundary/pull/5101 and https://github.com/hashicorp/boundary/pull/5107

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,12 +4,44 @@ Canonical reference for changes, improvements, and bugfixes for Boundary.
...
+### Deprecations/Changes
+
 * Remove deprecated `controllers` field from the worker config, which was deprecated in 0.9.0 for
 `initial_upstreams`([PR](https://github.com/hashicorp/boundary/pull/5125))
 
 ## 0.17.2 (Unreleased)
 
-### Changes
+### New and Improved
```
[the remark this answers is not in the document]
Comment (tmessi) [reply CHANGELOG.md:14]: Fixed.

File: CHANGELOG.md (hunk 1 above)
Comment (tmessi) [reply CHANGELOG.md]: Do you have a summary line for those? I'm not entirely sure what each one is doing. Is #5101 updating the client API to support pagination and then updating the client cache to use the pagination for targets? And then #5107 is updating the client cache to use this pagination for the remaining resources?

Comment (tmessi) [reply CHANGELOG.md]: I added these, let me know if that looks good.

**Review by jefferai — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jefferai) [reply CHANGELOG.md]: Looks good, thank you!

**Pull request merged**